docs/logs/getting-started-aspnetcore/FoodSupplyLogs.cs

```diff
@@ -0,0 +1,24 @@
+// <copyright file="FoodSupplyLogs.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+public static partial class FoodSupplyLogs
+{
+    [LoggerMessage(
+        EventId = 1,
+        Level = LogLevel.Information,
+        Message = "Food `{name}` price changed to `{price}`.")]
+    public static partial void FoodPriceChanged(this ILogger logger, string name, double price);
```

Comment:
Creating a new 25 line file for this one method feels very verbose given the ASP.NET Core minimal APIs were designed to embrace minimalism. I'd suggest defining this class in Program.cs to avoid the extra file. Other doc samples already do these types of inline class definitions ([example](https://learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis/responses?view=aspnetcore-7.0#configure-json-serialization-options-globally)). Starting in Microsoft.Extensions.Logging.Abstractions v8 EventId can be auto-generated from a hash of the method name and more stream-lined constructor overloads are available so you might reformat this as:

```C#
public static partial class FoodSupplyLogs
{
    [LoggerMessage(LogLevel.Information, "Food `{name}` price changed to `{price}`.")]
    public static partial void FoodPriceChanged(this ILogger logger, string name, double price);
}
```

If your sample isn't yet using version 8 (reasonable given its not quite out yet) then the current formatting is fine and I'd suggest putting it in Program.cs.

Reply:
Right, for hello world, we want to show simple logging, then we want to show how to get to this point when you "mature". I wouldn't start with this in the minimal API docs.

Reply:
@davidfowl are you suggesting that the OTel docs include multiple samples describing this progression and the reasoning behind when you choose to "mature"? I worry that will make the docs 2x longer than they are now and many devs are going to stop reading as soon as they have the first example that works. IMO the increment in complexity is small enough that if we had their attention long enough for only one example, I'd want them to see an example that uses LoggerMessage.

Reply:
I've made the updates based on our discussion, PTAL.

Reply:
I don't think it needs to be 2x bigger to show these examples

Reply:
You are right it can be done more succinctly if we use only small snippets rather than replicating the full example. I made a [comment](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4821/files#r1318516277) where I think that stuff could go.